### DISCUSSION TITLE
zsh: bad CPU type in executable: ./dnscrypt-proxy

### DISCUSSION THREAD

**Discussion by hapejadul** [DNSCrypt/dnscrypt-proxy#1710] [Q&A] [answered]
Im trying to install dnscrypt-proxy on my M1 macbook air following instruction from here (https://github.com/dnscrypt/dnscrypt-proxy/wiki/Installation-macOS), but when I type ./dnscrypt-proxy give me error zsh: bad CPU type in executable: ./dnscrypt-proxy so I cant continue, can someone help me how to fix this?

**Comment by jedisct1** [comment, marked as answer]
Hi!

What file did you download exactly?

**Reply by hapejadul** [reply]
dnscrypt-proxy-macos_arm64-2.0.45.zip

**Reply by jedisct1** [reply]
Thank you!

Indeed, it looks like that archive was for macOS Intel, in spite of the name.

It should be fixed now. Can you download it again? The file has the same name, but should now be a native executable for Apple Silicon.

**Reply by hapejadul** [reply]
Thank you it work now
